### ISSUE TITLE
Softmax and crossentropy are prone to produce NaNs

### ISSUE THREAD

**Issue by pevnak** [FluxML/NNlib.jl#68]
Hi,

I have found examples, where softmax or logitcrossentropy overflows. I will try to find the problem, but I will file the issue first.

This overflows logitcrossentropy. 
```
o = param(reshape([ -431.0;  279.0;   427.0], 3, 1))
y = [true, false, false]
f = Flux.logitcrossentropy(o, y);
Flux.Tracker.back!(f); 
Flux.Tracker.grad(o)
```
I believe the problem is in  `softmax` in this line `∇logsoftmax(Δ, xs) = ∇softmax(Δ ./ softmax(xs), xs)` in logsoftmax.jl:26.

A second example just overflows the softmax as
```
y = [true, false, false]
o = param(reshape([ -1047, -981, 1891],3,1))
Flux.Tracker.back!(sum(softmax(o))); 
Flux.Tracker.grad(o)
```
The problem here is that `softmax(o)` outputs `[0.0, 0.0, 1.0]`. The problem here is in  ∇softmax, which uses exp without checking that it might overflow.

**Comment by pevnak** [comment]
This is now fixed in #75 and can be closed after merging.
